pennylane/measurements.py

```diff
@@ -770,7 +770,7 @@ def circuit():
     if isinstance(op, qml.Hamiltonian):
         raise qml.QuantumFunctionError("Hamiltonians are not supported for rotating probabilities.")
 
-    if isinstance(op, (qml.ops.Sum, qml.ops.SProd)):  # pylint: disable=no-member
+    if isinstance(op, (qml.ops.Sum, qml.ops.SProd, qml.ops.Prod)):  # pylint: disable=no-member
```

Comment:
Once the `arithmetic_depth` property in #2835 we could instead check `if op.arithmetic_depth > 1`.

Reply:
Sounds good! Might be worth adding this fix into that PR as I imagine Prod will probably get merged first?